### ISSUE TITLE
IDE Support: Tracking Issue

### ISSUE THREAD

**Issue by Jamalam360** [pest-parser/pest-ide-tools#10]
This issue tracks progress on support for IDEs other than VSCode.

# Progress

## General Tasks

- [ ] Pull request and merge the rewrite to Tower LSP, allowing us to communicate over std i/o as defined via the LSP specification.
- [ ] Create an XML TextMate syntax - I haven't seen another editor that supports the JSON TextMate bundles that VSCode uses (maybe VSCode can use the XML ones so that there's only one source of syntax truth?).

## Editor List

- [ ] Sublime Text
- [ ] Helix
- [ ] (Neo)vim

**Comment by tomtau** [comment]
Emacs? I don't use it for Rust much though

**Comment by tomtau** [comment]
https://lapce.dev/ - I haven't tried this editor, but looks cool for Rust

**Comment by ItsAleph** [comment]
Getting working lsp integration in Helix is straightforward but without syntax highlighting it looks quite incomplete. Unfortunately, Helix only supports Tree-sitter grammars and not TextMate ones, so there should be a TS grammar to make full experience. Thanks for your work!

**Comment by Jamalam360** [comment]
That is unfortunate but I will look into making a tree-sitter grammar as well

**Comment by Jamalam360** [comment]
As far as I understand we can send a pull request to https://github.com/helix-editor/helix/blob/master/languages.toml to get the server added officially. @ItsAleph do you know if that is correct?

**Comment by ItsAleph** [comment]
Yes, the file you linked is the one controlling built-in support for language servers (though user can override these using user-scope `~/.config/helix/languages.toml` or project-scope `.helix/languages.toml`).

**Comment by ItsAleph** [comment]
> https://lapce.dev/ - I haven't tried this editor, but looks cool for Rust

Just tried it out and it looks like the plugin providing rust support suffers from several high-severity issues (up to not working *at all*, see [lapce/lapce-rust#24](https://github.com/lapce/lapce-rust/issues/24), which happened to me too). There isn't any active development happening on it either. Since one would likely want to work with both pest and rust in the same editor, there isn't much reason to put effort into creating a plugin for right now.

**Comment by Jamalam360** [comment]
Thank you for bringing this up

**Comment by wale** [comment]
Update the [IntelliJ plugin](https://github.com/pest-parser/intellij-pest/) to use the LSP?

**Comment by Jamalam360** [comment]
No, probably not going to happen unfortunately. Intellij IDEs (other than the upcoming Fleet) don't support the LSP natively.

**Comment by Jamalam360** [comment]
Good news! This news just came out today, the future is looking good!

https://blog.jetbrains.com/idea/2023/06/intellij-idea-2023-2-eap-7/

**Comment by tomtau** [comment]
Nice https://github.com/JetBrains/intellij-plugins/tree/master/prisma/src/org/intellij/prisma/ide/lsp

**Comment by TiemenSch** [comment]
Would you mind publishing the VSX extension to the https://open-vsx.org/ registry as well?
That would enable VSCodium support :)

**Comment by tomtau** [comment]
> Would you mind publishing the VSX extension to the https://open-vsx.org/ registry as well?
> That would enable VSCodium support :)

@tiemensch I manually published it here: https://open-vsx.org/extension/pest/pest-ide-tools for the time being, but it should be easy to add it to the existing automated pipeline for new versions later

**Comment by Jamalam360** [comment]
CI set up for OpenVSX

**Comment by Jamalam360** [comment]
https://github.com/helix-editor/helix/pull/7878

**Comment by inferiorhumanorgans** [comment]
FWIW I've created a rough tree-sitter parser here:

https://git.sr.ht/~az1/tree-sitter-pest

**Comment by TheBlckbird** [comment]
What about [Zed](https://zed.dev/)?

**Comment by Jamalam360** [comment]
Yeah, we'll just have to wait and then I imagine it will be similar to (neo)vim in terms of what's needed from us

**Comment by TheBlckbird** [comment]
@tomtau Thank you!

**Comment by huacnlee** [comment]
@tomtau Done

https://github.com/zed-industries/extensions/pull/354/

**Comment by erasin** [comment]
https://github.com/helix-editor/helix/pull/10616 helix supported
